### PR TITLE
FWF-3739:[Feature]- Save and publish. with confirmation modals.

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-3739
Issue Type: FEATURE

# Changes
Save and publish

# Screenshots (if applicable)
image
image
image
image
image

### CODE REVIEW

**Review by shuhaib-aot — APPROVED** [review]

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 1]
```diff
@@ -494,69 +518,62 @@ const Edit = React.memo(() => {
...
+    closeModal();
+    setIsPublishLoading(true);
+    unPublish(processListData.id)
+      .then(() => {
+        setPromptNewVersion(true);
+        setIsPublishLoading(false);
+        setIsPublished(!isPublished);
+      })
+      .catch((err) => {
+        setIsPublishLoading(false);
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:531]: can we move setIsPublishLoading to finally() ? @Josephalexantony-aot

File: forms-flow-web/src/components/Modals/NewVersionModal.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    }, [acceptState]);
+
+    return (
+        <Modal show={show} onHide={clearState} dialogClassName="modal-50w" size="sm" centered={true}>
+            <Modal.Header>
+                <Modal.Title>
+                    <b>{title}</b>
+                </Modal.Title>
+                <div className="d-flex align-items-center">
+                    <CloseIcon width="16.5" height="16.5" onClick={() => {
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/NewVersionModal.js:58]: add data-testid

File: forms-flow-web/src/components/Modals/NewVersionModal.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            <Modal.Body className="newversion-modal-body">
+                <div className="d-flex justify-content-between align-items-center">
+                    <label className="form-check-label" htmlFor="acceptFirst">
+                        Will be saved as a Version {newVersion}
+                    </label>
+                    <div className="d-flex align-items-center">
+                        <input
+                            className="form-check-input"
+                            type="checkbox"
+                            id="acceptFirst"
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/NewVersionModal.js:72]: add data-testid

File: forms-flow-web/src/components/Modals/NewVersionModal.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+                <div className="d-flex justify-content-between align-items-center">
+                    <label className="form-check-label" htmlFor="acceptSecond">
+                        You will not be able to affect previous submissions after
+                    </label>
+                    <div className="d-flex align-items-center">
+                        <input
+                            className="form-check-input"
+                            type="checkbox"
+                            id="acceptSecond"
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/NewVersionModal.js:88]: add data-testid

File: forms-flow-web/src/components/Modals/NewVersionModal.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+
+                <div className="d-flex justify-content-between align-items-center">
+                    <label className="form-check-label" htmlFor="acceptThird">
+                        This action cannot be undone
+                    </label>
+                    <div className="d-flex align-items-center">
+                        <input
+                            className="form-check-input"
+                            type="checkbox"
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/NewVersionModal.js:104]: add data-testid

**Review by arun-s-aot — APPROVED** [review]

**Pull request merged**